### ISSUE TITLE
Always auto-scroll Interactive Tab before reaching the bottom 

### ISSUE THREAD

**Issue by 205g0** [microsoft/vscode-jupyter#4139]
Related to https://github.com/microsoft/vscode-jupyter/issues/4043 but  more convenient: once input or output reaches 85% of vertical pane height or 15% from bottom auto-scroll the rendered cell blocks in a way that there is 15% whitespace at the bottom:

Why? At the moment, it scrolls sometimes but sometimes not which is the first annoyance (it's unpredictable) . And when it scrolls, it scrolls in a way that the last block (in a full screen) is very close to the bottom: 

![image](https://user-images.githubusercontent.com/74575852/101301858-5a596a00-383a-11eb-8677-d3a0b29dc365.png)

Latter just feels uncomfortable because you often don't know if it scrolled and it feels cramped. Even the scrollbar on the right isn't a good indication if you are the end or not. It doesn't looks like.

In other words, the 15% whitespace would indicate that the Interactive Tab is really at the bottom. Re the 15%: This is to be discussed, it should be a healthy trade-off between knowing I am at the end and seeing enough of the last rendered cell blocks.

**Comment by joyceerhl** [comment]
@205g0, we have a setting to auto-scroll the IW to show the output of the last statement executed. Do you have this in your settings.json?
```
"jupyter.alwaysScrollOnNewCell": true 
```

If the above setting is false or not set, we only autoscroll to show the output of the last executed statement if the bottom of the previous cell is visible (which may be the cause of the unpredictability that you're reporting).

**Comment by 205g0** [comment]
@joyceerhl Thanks for that hint and good to know! I had it off and just turned it on.

What's nice now: It's scrolling *and* going to the bottom even when I am somewhere in the middle of the document.

But there are still issues:

- It's sometimes cutting off the output, I couldn't find out when but often when showing errors:

![image](https://user-images.githubusercontent.com/74575852/101464199-a2fa4b80-393e-11eb-9a35-d0eac92e10a0.png)

- It's not autoscrolling when the cell is pure markdown; this is minor because it's predictably not scrolling in this case; still it would be nice because yeah, I often do this quickly just to mark where I am and hence would like to see this in the log as well and markdown in the log is more readable
- Then, I would still kindly suggest to introduce a small gap to the bottom to make it even more clear, hey your are at the bottom; this is minor as well and also a matter of taste

**Comment by DonJayamanne** [comment]
Closing this issue as I beliebe this has been resolved in the latest Interactive Window (which happens to be a first class cititzen of VS COde).
If the issue persists, please feel free to re-open this issue or create a new one.
